tikv-client/src/main/java/com/pingcap/tikv/util/KeyRangeUtils.java

```diff
@@ -127,18 +127,17 @@ public static KeyRange makeCoprocRange(Range<Key> range) {
       return ranges;
     }
 
-    RangeSet<Key> rangeSet = TreeRangeSet.create();
-    for (KeyRange keyRange : ranges) {
-      Range<Key> range = makeRange(keyRange.getStart(), keyRange.getEnd());
-      rangeSet.add(range);
-    }
+    // Merge discrete ranges into one complete big range
+    ByteString start = ranges.get(0).getStart();
+    ByteString end = ranges.get(ranges.size() - 1).getEnd();
```

Comment:
It is guaranteed `ranges` list contains at least 2 elements, see logic at the beginning of this function.

Reply:
maybe better to add it into comment

Reply:
I think this strategy has worst performance concerning a couple of random small ranges

Reply:
It's fine to leave it a big big range for now. Fine tuning needs a lot of consideration and might be in vain since we do not know the distance of handle from each other and detail cost balance of reading.